src/west/commands/project.py

```diff
@@ -208,6 +193,49 @@ def do_run(self, args, user_args):
             _git(project, 'status', extra_args=user_args)
 
 
+class Update(WestCommand):
+    # Commit comment
+    def __init__(self):
+        super().__init__(
+            'update',
+            'Update projects descibed in west.yml',
+            _wrap('''
+            Updates all projects according to the manifest file, `west.yml`,
+            in the manifest repository.
+            All projects will be checked out in detached HEAD and the local
+            manifest-rev branch will be reset hard to same HEAD.
+
+            Any local branches will not be altered by this command.
+
+            This command will not touch the manifest repository.
+            '''))
+
+    def do_add_parser(self, parser_adder):
+        return _add_parser(parser_adder, self, _no_update_arg,
+                           _project_list_arg)
+
+    def do_run(self, args, user_args):
+        if args.update:
+            _update_west()
+
+        for project in _projects(args, listed_must_be_cloned=False,
```

Comment:
Shouldn't this be `listed_must_be_cloned=True`? The first thing we do is take the current branch of the project, so it must be cloned for this to work.

Reply:
OK, sounds good.